source/server/server.cc

```diff
@@ -530,7 +530,7 @@ void InstanceImpl::initialize(Network::Address::InstanceConstSharedPtr local_add
                                      : ReusePortDefault::False;
```

Comment:
A question, does it mean the user never can upgrade to the true default value by hot restart? So that means we should keep those code forever?

Reply:
@mattklein123 this question probably needs your help. I thought we should remove all the upgrade code. But from the code and comments, I understand the hot restart can't change the default value. That means we only can remove those codes when all the users have a cold restart. Then probably this code never can be deleted?

Reply:
We don't have a lot of precedent for this. I think technically we could likely remove the code when the last version that had the old default falls out of support. You could leave a comment/TODO about this and we can circle back later? cc @envoyproxy/envoy-maintainers